pandas/core/indexes/base.py

```diff
@@ -6364,7 +6364,7 @@ def map(self, mapper, na_action=None):
 
         # we can return a MultiIndex
         if new_values.size and isinstance(new_values[0], tuple):
-            if isinstance(self, MultiIndex):
+            if isinstance(self, MultiIndex) and len(new_values[0]) == self.nlevels:
```

Comment:
what happens if its a Multiindex bu tthis is not true? do we have a test case for that

Reply:
The checks that happen here are only to determine the `names` of the new index. 

```python
if isinstance(self, MultiIndex) and len(new_values[0]) == self.nlevels:
     names = self.names
 elif self.name:
    names = [self.name] * len(new_values[0])
else:
  names = None
```

If the old index is a `MultiIndex` then we reuse  `names` for the new index. However, if the number of levels changed then `len(self.names) != len(len(new_values[0])` and an error will be raised. With the change, `names` is not preserved and reset as seen in the snippet. The last of the three tests that I added checks for the new case.